src/commands/test.js

```diff
@@ -22,49 +26,96 @@ module.exports = {
     let { print } = toolbox
 
     // Read CLI parameters
-    let { f, force, h, help, l, logs, v, version } = toolbox.parameters.options
-    let datasource = toolbox.parameters.first
+    let {
+      c,
+      coverage,
+      d,
+      docker,
+      f,
+      force,
+      h,
+      help,
+      l,
+      logs,
+      v,
+      version,
+    } = toolbox.parameters.options
 
     // Support both long and short option variants
-    force = force || f
-    help = help || h
-    logs = logs || l
-    version = version || v
+    let coverage_opt = coverage || c
+    let docker_opt = docker || d
+    let force_opt = force || f
+    let help_opt = help || h
+    let logs_opt = logs || l
+    let version_opt = version || v
+
+    // Fix if a boolean flag (e.g -c, --coverage) has an argument
+    try {
+      fixParameters(toolbox.parameters, {
+        h,
+        help,
+        c,
+        coverage,
+        d,
+        docker,
+        f,
+        force,
+        l,
+        logs
+      })
+    } catch (e) {
+      print.error(e.message)
+      process.exitCode = 1
+      return
+    }
+
+    let datasource = toolbox.parameters.first || toolbox.parameters.array[0]
 
     // Show help text if requested
-    if (help) {
+    if (help_opt) {
       print.info(HELP)
       return
     }
 
-    const platform = getPlatform(logs)
-    if (!version) {
-      let result = await fetch('https://api.github.com/repos/LimeChain/matchstick/releases/latest')
-      let json = await result.json()
-      version = json.tag_name
+    let result = await fetch('https://api.github.com/repos/LimeChain/matchstick/releases/latest')
+    let json = await result.json()
+    let latest_version = json.tag_name
+
+    if(docker_opt) {
+      runDocker(coverage_opt, datasource, version_opt, latest_version, print)
+    } else {
+      runBinary(coverage_opt, datasource, force_opt, logs_opt, version_opt, latest_version, print)
     }
+  }
+}
 
-    const url = `https://github.com/LimeChain/matchstick/releases/download/${version}/${platform}`
+async function runBinary(coverage_opt, datasource, force_opt, logs_opt, version_opt, latest_version, print) {
+  const platform = getPlatform(logs_opt)
 
-    if (logs) {
-      console.log(`Download link: ${url}`)
-    }
+  const url = `https://github.com/LimeChain/matchstick/releases/download/${version_opt || latest_version}/${platform}`
 
-    let binary = new Binary(platform, url, version)
-    await binary.install(force)
-    datasource ? binary.run(datasource) : binary.run()
+  if (logs_opt) {
+    console.log(`Download link: ${url}`)
   }
+
+  let binary = new Binary(platform, url, version_opt)
+  force_opt ? await binary.install(true) : await binary.install(false)
+  let args = ""
+
+  datasource ? args = datasource : args
+  coverage_opt ? args = args + ' -c' : args
+  args !== '' ? binary.run(args.trim()) : binary.run()
```

Comment:
I'm okay with the last ternary statement, but I don't think they make sense in the first two lines, because you're not using the value returned in the `else` case (`: args`).

I'm not the biggest fan of inline `if`s but since you're doing one liners I think that's better than the ternary one, like this:

```javascript
  if (datasource) args = datasource
  if (coverage_opt) args = args + ' -c'
  args !== '' ? binary.run(args.trim()) : binary.run()
```


What do you think?